new code

```diff
@@ -0,0 +1,163 @@
+                System.out.print("\t");
+                for(l=0 ; l<3 ; l++)
+                    {
+                        System.out.print(" " + k + " ");
+                        k++;
+                    }
+                System.out.println();
+            }
+            int winflag = 0;
+            int flag = 0;
+            int r,c;
+            int ctr = 0;
+            while(ctr < 9)  
+            {
+                System.out.println();
+                System.out.print("enter number according to the matrix beside the game: ");
+                m = sc.nextInt();
+                c = (m-1)%3;
+                r = (m-1)/3;
+                if(m > 9 || m < 1)
+                {
+                    System.out.print("Please enter numbers from 1-9 ");
+                    ctr--;
+                    System.out.println();
+                }
+                else
+                {
+                    if(mat[r][c] == " x " || mat[r][c] == " o ")
+                    {
+                        System.out.print("player has already entered in that position. Please enter somewhere else!");
+                        System.out.println();
+                        ctr--;
+                    }
+                    else
+                    {
+                        if (flag == 0)
+                        {
+                            mat[r][c] = " x ";
+                            flag = 1;
+                        }
+                        else if (flag == 1)
+                        {
+                            mat[r][c] = " o ";
+                            flag = 0;
+                        }
+                    }
+                }
+
+                if (mat[0][0] == mat[1][0] && mat[1][0] == mat[2][0] && mat[2][0] != " - ")
+                {
+                    winflag = 1;
+                
+                }
+                if (mat[0][1] == mat[1][1] && mat[1][1] == mat[2][1] && mat[2][1] != " - ")
+                {
+                    winflag = 1;
+                }
+                if (mat[0][2] == mat[1][2] && mat[1][2] == mat[2][2] && mat[2][2] != " - ")
+                {
+                    winflag = 1;
+                }
+                if (mat[1][0] == mat[1][1] && mat[1][1] == mat[1][2] && mat[1][2] != " - ")
+                {
+                    winflag = 1;
+                }
+                if (mat[2][0] == mat[2][1] && mat[2][1] == mat[2][2] && mat[2][2] != " - ")
+                {
+                    winflag = 1;
+                }
+                if (mat[0][0] == mat[1][1] && mat[1][1] == mat[2][2] && mat[2][2] != " - ")
+                {
+                    winflag = 1;
+                }
+                if (mat[0][2] == mat[1][1] && mat[1][1] == mat[2][0] && mat[2][0] != " - ")
+                {
+                    winflag = 1;
+                }
+                if (mat[0][0] == mat[0][1] && mat[0][1] == mat[0][2] && mat[0][0] != " - ")
+                {
+                    winflag = 1;
```

Comment:
move all these if conditions to a function 
the function can return a flag (boolean) 
true if won else false